File: src/access-control/policies/alwaysAllowAccessPolicy.ts
import { JSONObject } from "../../common/jsonTypes.js";
import { Document } from "../../document/document.js";
import { AccessIdentity } from "../accessIdentity.js";
import { ResourceAccessPolicy } from "../resourceAccessPolicy.js";

export class AlwaysAllowAccessPolicy implements ResourceAccessPolicy {
  policy: string = "AlwaysAllowAccessPolicy";
  policyJSON: JSONObject = {};

  async testDocumentReadPermission(
    _document: Document,
    _requestor?: AccessIdentity,
  ): Promise<boolean> {
    return true;
  }

  async testPolicyPermission(
    _requestor: AccessIdentity,
  ): Promise<string[] | boolean> {
    return true;
  }
}

File: python/src/semantic_retrieval/access_control/policies/always_allow_access_policy.py
from typing import List, Optional, Union

from semantic_retrieval.access_control.access_identity import AccessIdentity

# from semantic_retrieval.access_control.policies.resource_access_policy import ResourceAccessPolicy
from semantic_retrieval.access_control.resource_access_policy import (
    ResourceAccessPolicy,
)
from semantic_retrieval.document.document import Document


class AlwaysAllowAccessPolicy(ResourceAccessPolicy):
    policy: str = "always_allow"

    async def testDocumentReadPermission(
        self, document: Document, requestor: Optional[AccessIdentity] = None
    ) -> bool:
        return True

    async def testPolicyPermission(
        self, requestor: AccessIdentity
    ) -> Union[List[str], bool]:
        return True
